snc-library.sh

```diff
@@ -283,8 +283,15 @@ function wait_till_cluster_stable() {
     done
 
     # Wait till all the pods are either running or complete state
+    # Add 5 mins timeout
+    runtime="5 minute"
+    endtime=$(date -ud "$runtime" +%s)
     while ${OC} get pod --no-headers --all-namespaces | grep -v Running | grep -v Completed; do
        sleep 2
+       if [[ $(date -u +%s) -ge $endtime ]]; then
+	   echo "Not able to get all the containers in running/completed state"
+	   exit 1
+       fi
```

Comment:
@cfergeau retry where?

Reply:
This `while` loop is relatively close to `retry` implementation, could `retry` be used instead of that loop?

Reply:
Still wondering if this could be something like:
```
function all_pods_running_or_completed {
    # not fully sure what this what needs to be done here so that retry() considers this failed/succeess
        ${OC} get pod --no-headers --all-namespaces | grep -v Running | grep -v Completed
}

    retry all_pods_running_or_completed
```

Reply:
@cfergeau in this scenario `${OC} get pod --no-headers --all-namespaces | grep -v Running | grep -v Completed` should return non zero value as soon as all the containers are in running/completed state so initial phase of retry is going to successful. What we actually want `all_pods_running_or_completed` function shouldn't successful until all the pods are in running or completed state so it should be `! ${OC} get pod --no-headers --all-namespaces | grep -v Running | grep -v Completed`